server/src/transport/ws.rs

```diff
@@ -310,7 +310,6 @@ pub(crate) async fn background_task<L: Logger>(sender: Sender, mut receiver: Rec
 			max_log_length,
 			max_response_body_size,
 			sink: sink.clone(),
-			sink_permit,
```

Comment:
without "the reserve" we will read the underlying socket and then the client is not "forced" to read to send new messages.

that's is the reason of the reserve entirely

Reply:
otherwise, it will not propagated all the way down to the TCP level when the window size is adjusted etc

Reply:
I get your concern, this permit could interfere with the stuff trying to send stuff, so we could do:

```rust
		// Wait until there is a slot in the bounded channel.
		//
		// This will force the client to read socket on the other side
		// otherwise the socket will not be read again.
		let shutdown = loop {
			if stop_handle.shutdown_requested() {
				break true;
			}

			if sink.capacity() != 0 {
				break false;
			}
		};

		if shutdown {
			break Ok(Shutdown::Stopped);
		}

```